File: mobile/src/domain/entities/social.ts
export interface PublicUserSummary {
  id: number;
  username: string;
  display_name: string | null;
  avatar_url: string | null;
  is_following: boolean;
}

export interface PublicUserProfile extends PublicUserSummary {
  bio: string | null;
  created_at: string;
  followers_count: number;
  following_count: number;
  reviews_count: number;
  watch_logs_count: number;
}

export interface GenreStat {
  name: string;
  count: number;
}

export interface PublicUserStats {
  watched_count: number;
  estimated_hours: number;
  top_genres: GenreStat[];
  average_rating: number | null;
}

export interface FavoriteMediaItem {
  position: number;
  media: {
    tmdb_id: number;
    media_type: 'movie' | 'tv';
    title: string;
    poster_path: string | null;
    vote_average: number;
    release_date: string | null;
  };
}

export interface FavoriteMediaWriteItem {
  position: number;
  tmdb_id: number;
  media_type: 'movie' | 'tv';
}

export interface ActivityActor {
  id: number;
  username: string;
  display_name: string | null;
  avatar_url: string | null;
}

export interface ReviewActivity {
  title: string;
  poster_path: string | null;
  id: number;
  activity_type: 'review';
  created_at: string;
  actor: ActivityActor;
  review_id: number;
  tmdb_id: number;
  media_type: 'movie' | 'tv';
  rating: number;
  body_preview: string;
  contains_spoilers: boolean;
}

export interface WatchLogActivity {
  title: string;
  poster_path: string | null;
  id: number;
  activity_type: 'watch_log';
  created_at: string;
  actor: ActivityActor;
  watch_log_id: number;
  tmdb_id: number;
  media_type: 'movie' | 'tv';
  watched_at: string;
  rating: number | null;
}

export interface FollowActivity {
  id: number;
  activity_type: 'follow';
  created_at: string;
  actor: ActivityActor;
  followed_user: PublicUserSummary;
}

export interface ListCreatedActivity {
  id: number;
  activity_type: 'list_created';
  created_at: string;
  actor: ActivityActor;
  list_id: number | null;
  list_name: string | null;
  items_count: number;
  is_public: boolean;
}

export type ActivityItem = ReviewActivity | WatchLogActivity | FollowActivity | ListCreatedActivity;

export interface FeedPage {
  items: ActivityItem[];
  next_cursor: string | null;
}

export interface VisualFeedParticipant {
  id: number;
  username: string;
  display_name: string | null;
  avatar_url: string | null;
  activity_type: 'review' | 'watch_log';
  rating: number | null;
  created_at: string;
}

export interface VisualFeedItem {
  media: {
    tmdb_id: number;
    media_type: 'movie' | 'tv';
    title: string;
    poster_path: string | null;
    vote_average: number;
    release_date: string | null;
  };
  participants: VisualFeedParticipant[];
  recent_activity_count: number;
  latest_activity_at: string;
}

File: backend/app/domain/entities/social.py
from dataclasses import dataclass
from datetime import date, datetime

from app.domain.entities.media import MediaItem


@dataclass
class PublicUserSummary:
    id: int
    username: str
    display_name: str | None
    avatar_url: str | None
    is_following: bool


@dataclass
class PublicUserProfile:
    id: int
    username: str
    display_name: str | None
    bio: str | None
    avatar_url: str | None
    created_at: datetime
    followers_count: int
    following_count: int
    reviews_count: int
    watch_logs_count: int
    is_following: bool


@dataclass
class GenreStat:
    name: str
    count: int


@dataclass
class PublicUserStats:
    watched_count: int
    estimated_hours: float
    top_genres: list[GenreStat]
    average_rating: float | None


@dataclass
class FavoriteMediaSelection:
    position: int
    tmdb_id: int
    media_type: str


@dataclass
class ActivityActor:
    id: int
    username: str
    display_name: str | None
    avatar_url: str | None


@dataclass
class FeedCursor:
    created_at: datetime
    activity_id: int


@dataclass
class FollowedUser:
    id: int
    username: str
    display_name: str | None
    avatar_url: str | None


@dataclass
class BaseActivity:
    id: int
    activity_type: str
    created_at: datetime
    actor: ActivityActor


@dataclass
class ReviewActivity(BaseActivity):
    review_id: int
    tmdb_id: int
    media_type: str
    rating: int
    body_preview: str
    contains_spoilers: bool


@dataclass
class WatchLogActivity(BaseActivity):
    watch_log_id: int
    tmdb_id: int
    media_type: str
    watched_at: date
    rating: int | None


@dataclass
class FollowActivity(BaseActivity):
    followed_user: FollowedUser


@dataclass
class ListCreatedActivity(BaseActivity):
    list_id: int | None
    list_name: str | None
    items_count: int
    is_public: bool


@dataclass
class VisualFeedParticipant:
    id: int
    username: str
    display_name: str | None
    avatar_url: str | None
    activity_type: str
    rating: int | None
    created_at: datetime


@dataclass
class VisualFeedItem:
    media: MediaItem
    participants: list[VisualFeedParticipant]
    recent_activity_count: int
    latest_activity_at: datetime


@dataclass
class RecentWatchLogEntry:
    id: int
    tmdb_id: int
    media_type: str
    watched_at: date
    rating: int | None
    created_at: datetime
    media: MediaItem
